sparse_ho/algo/implicit.py

```diff
@@ -14,45 +12,112 @@ class Implicit():
 
```

Comment:
Can you add a 3-lines explanation of the algo, so that it's clear which linear system we talk about in the rest of the docstring ?   Smth like "We first solve the inne rproblem, then find the Jac by solving a linear system on the support of the sol"